src/mongo/MongoScrapbook.ts

```diff
@@ -347,37 +347,68 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 			this.addErrorToCommand(err, ctx);
 			return {};
 		}
-		let argumentContextArray: mongoParser.ArgumentContext[] = filterType(argumentsToken.children, mongoParser.ArgumentContext);
 
-		let functionMap = { "ObjectId": this.objectIdToObject, "ISODate": this.dateToObject, "Date": this.dateToObject };
+		let argumentContextArray: mongoParser.ArgumentContext[] = filterType(argumentsToken.children, mongoParser.ArgumentContext);
 		if (argumentContextArray.length > 1) {
 			let err: IParsedError = parseError(`Too many arguments. Expecting 0 or 1 argument(s) to ${constructorCall}`);
 			this.addErrorToCommand(err, ctx);
 			return {};
 		}
-		if (constructorCall.text in functionMap) {
-			let args = [ctx, argumentContextArray.length ? argumentContextArray[0].text : undefined];
-			return functionMap[constructorCall.text].apply(this, args);
+
+		const tokenText: string | undefined = argumentContextArray.length ? argumentContextArray[0].text : undefined;
+		switch (constructorCall.text) {
+			case 'ObjectId':
+				return this.objectIdToObject(ctx, tokenText);
+			case 'ISODate':
+				return this.isodateToObject(ctx, tokenText);
+			case 'Date':
+				return this.dateToObject(ctx, tokenText);
+			default:
+				let unrecognizedNodeErr: IParsedError = parseError(`Unrecognized node type encountered. Could not parse ${constructorCall.text} as part of ${child.text}`);
+				this.addErrorToCommand(unrecognizedNodeErr, ctx);
+				return {};
+		}
+	}
+
+	private dateToObject(ctx: mongoParser.ArgumentContext | mongoParser.PropertyValueContext, tokenText?: string): { $date: Date | {} } {
+		let stringDate: Date | {} = this.tryToConstructDate(ctx, tokenText);
+		if (stringDate instanceof Date) {
+			return { $date: stringDate.toISOString() };
```

Comment:
Previously it did `toString` instead of `toISOString`. Honestly I'm not sure what's the expected behavior here, but I would err on the side of keeping it consistent with previous releases